### PR TITLE
Enhance data serializer

### PR DESCRIPTION
**Pull request by varas**
Enhance data serializer by removing non required `fmt` dependency and avoiding extra byte > string > byte conversion.

Also adds lacking map `make` on Integration constructor.

### Author

- [x] add a risk label after carefully considering the "blast radius" of your changes
- [x] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [x] assign at least one reviewer

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] ask questions about anything that isn't clear and obvious
- [ ] approve the PR when you consider it's good to merge

### CODE REVIEW

File: sdk/v2/data.go [hunk 1]
```diff
@@ -185,7 +184,7 @@ func (integration *Integration) Clear() {
 
 // toJSON returns the integration as a JSON string. If the pretty attribute is
```
Comment (areina) [inline sdk/v2/data.go]: super minor, but maybe we should change the comment about returning a string.

File: sdk/v2/data.go [hunk 2]
```diff
@@ -183,9 +182,9 @@ func (integration *Integration) Clear() {
 	integration.Data = []*EntityData{} // empty array preferred instead of null on marshaling.
 }
 
-// toJSON returns the integration as a JSON string. If the pretty attribute is
-// set to true, the JSON will be idented for easy reading.
-func (integration *Integration) toJSON(pretty bool) (string, error) {
+// toJSON serializes integration as JSON. If the pretty attribute is
+// set to true, the JSON will be indented for easy reading.
+func (integration *Integration) toJSON(pretty bool) ([]byte, error) {
```
Comment (smoya) [inline sdk/v2/data.go]: In a next iteration we could call this method `MarshalJSON` following the JSON Marshaler Interface from stdlib https://golang.org/pkg/encoding/json/#Marshaler

File: sdk/v2/data.go [hunk 3]
```diff
@@ -196,12 +195,12 @@ func (integration *Integration) toJSON(pretty bool) (string, error) {
 	}
 
 	if err != nil {
-		return "", fmt.Errorf("error marshalling to JSON: %s", err)
+		return []byte(""), errors.Errorf("error marshalling to JSON: %s", err)
```
Comment (smoya) [inline sdk/v2/data.go]: Do not create a new slice of byte but return `output` var directly

Comment (varas) [reply sdk/v2/data.go]: Why? That'd change the current behaviour

**Pull request merged**